lib/src/runner/configuration/args.dart

```diff
@@ -92,6 +92,11 @@ final ArgParser _parser = (() {
           'to provide improved test performance but at the cost of\n'
           'debuggability.',
       defaultsTo: true);
+  parser.addFlag("retry",
+      help: 'Retry failing tests if configured to do so.\n'
```

Comment:
Once you change the flag to `--no-retry`, make this "Don't re-run tests that have retry set."